packages/xo-web/src/xo-app/settings/config/xo-cloud-config/index.js

```diff
@@ -0,0 +1,131 @@
+
+const CloudConfig = decorate([
+  addSubscriptions({
+    cloudXoConfig: subscribeCloudXoConfig,
+    cloudXoConfigBackups: subscribeCloudXoConfigBackups,
+  }),
+  provideState({
+    initialState: () => ({ config: undefined }),
+    effects: {
+      downloadCloudXoConfig:
+        () =>
+        ({ config, isConfigDefined }) => {
+          if (isConfigDefined) {
+            window.open(config.content, '_blank')
+          }
+        },
+      uploadCloudXoConfig:
+        () =>
+        async ({ config, isConfigDefined }) => {
+          if (isConfigDefined) {
+            const resp = await fetch(`./rest/v0/cloud/xo-config/backups/${config.id}/actions/import?sync`, {
+              method: 'POST',
+            })
+            if (!resp.ok) {
+              throw new Error(resp.statusText)
+            }
+            return {
+              config: undefined,
+            }
+          }
+        },
+      onChangeCloudXoConfig: (_, config) => ({
+        config,
+      }),
+      toggleEnableCloudXoConfig:
+        () =>
+        async (state, { cloudXoConfig }) => {
+          let passphrase
+          if (!cloudXoConfig.enabled) {
+            const params = await confirm({
+              icon: 'backup',
+              title: _('cloudXoConfig'),
+              body: <BackupXoConfigModal />,
+            })
+            passphrase = params.passphrase
+          }
+
+          const resp = await fetch('./rest/v0/cloud/xo-config', {
+            method: 'PATCH',
+            headers: {
+              'Content-Type': 'application/json',
+            },
+            body: JSON.stringify({
+              enabled: !cloudXoConfig.enabled,
+              passphrase,
+            }),
+          })
+          if (!resp.ok) {
+            throw new Error(resp.statusText)
+          }
+          subscribeCloudXoConfig.forceRefresh()
+        },
+    },
+    computed: {
+      applianceId: async () => {
+        const { id } = await getApiApplianceInfo()
+        return id
+      },
+      groupedConfigs: ({ applianceId, sortedConfigs }) =>
+        sortBy(groupBy(sortedConfigs, 'xoa_id'), config => (config[0].xoa_id === applianceId ? -1 : 1)),
+      isConfigDefined: ({ config }) => config != null,
+      sortedConfigs: (_, { cloudXoConfigBackups }) =>
+        cloudXoConfigBackups?.sort((config, nextConfig) => config.created_at - nextConfig.created_at),
+    },
+  }),
+  injectState,
+  ({ effects, state, cloudXoConfig }) => (
+    <div>
+      <div className='mb-1'>
+        <em>
+          <Icon icon='info' /> {_('xoConfigCloudBackupTips')}
+        </em>
+        <br />
+        <ActionButton
+          btnStyle={cloudXoConfig?.enabled ? 'warning' : 'primary'}
+          handler={effects.toggleEnableCloudXoConfig}
+          icon='backup'
+        >
+          {cloudXoConfig?.enabled ? _('disableXoCloudBackup') : _('enableXoCloudBackup')}
+        </ActionButton>
+      </div>
+      <div>
+        <h2>{_('xoConfig')}</h2>
```

Comment:
In terms of page layout, I think it would be more readable like this:
- Tab: XO Config (to contrast with Cloud Config)
- Page:
	- XO Config Cloud Backup
		- Enable/Disable recurring backup
			- button
		- Download
			- select
			- button
		- Upload (or backup?)
			- button
	- Local
		- Import
			- dropzone
		- Export
			- button

/cc @olivierlambert

Reply:
You're right. It's related to the fact that I didn't understand the purpose of the Import (aka Restore) button. I edited my comment :)